Spigot-API-Patches/0291-PlayerMoveEvent-Improvements.patch

```diff
@@ -0,0 +1,46 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: HexedHero <6012891+HexedHero@users.noreply.github.com>
+Date: Thu, 29 Apr 2021 10:31:44 +0100
+Subject: [PATCH] PlayerMoveEvent Improvements
+
+
+diff --git a/src/main/java/org/bukkit/event/player/PlayerMoveEvent.java b/src/main/java/org/bukkit/event/player/PlayerMoveEvent.java
+index 1a58734d919fae247eeb85dd785fd59990856505..26416cd1cea4038528335826697a6ff9f3033dfe 100644
+--- a/src/main/java/org/bukkit/event/player/PlayerMoveEvent.java
++++ b/src/main/java/org/bukkit/event/player/PlayerMoveEvent.java
+@@ -93,6 +93,35 @@ public class PlayerMoveEvent extends PlayerEvent implements Cancellable {
+         this.to = to;
+     }
+ 
++    // Paper start - PlayerMoveEvent improvements
++    /**
++     * Check if the player has changed position in the event
++     * 
++     * @return whether the player has changed position or not
++     */
++    public boolean hasChangedPosition() {
++        return from.getX() != to.getX() || from.getY() != to.getY() || from.getZ() != to.getZ();
++    }
++
++    /**
++     * Check if the player has changed orientation in the event
++     * 
++     * @return whether the player has changed orientation or not
++     */
++    public boolean hasChangedOrientation() {
++        return from.getPitch() != to.getPitch() || from.getYaw() != to.getYaw();
++    }
++
++    /**
++     * Check if the player has changed to a new block in the event
++     * 
++     * @return whether the player has changed block or not
++     */
++    public boolean hasChangedBlock() {
++        return !from.getBlock().equals(to.getBlock());
```

Comment:
Maybe a comparision with the `getBlockX/Y/Z` would be more optimized to prevent this kind of allocation ?

Reply:
I'll change that now, thank you :D